src/main/java/uk/gov/pay/api/model/directdebit/agreement/AgreementError.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.pay.api.model.directdebit.agreement;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+import static com.google.common.collect.ObjectArrays.concat;
+import static java.lang.String.format;
+
+@ApiModel(value = "AgreementError", description = "An Agreement Error response")
+@JsonInclude(NON_NULL)
+public class AgreementError {
+
+    public enum Code {
+
+        CREATE_AGREEMENT_ACCOUNT_ERROR("P0199", "There is an error with this account. Please contact support"),
+        CREATE_AGREEMENT_CONNECTOR_ERROR("P0198", "Downstream system error"),
+        
+        GET_AGREEMENT_NOT_FOUND_ERROR("P0200", "Not found"),
+        GET_AGREEMENT_CONNECTOR_ERROR("P0298", "Downstream system error"),
+
+        TOO_MANY_REQUESTS_ERROR("P0900", "Too many requests"),
+        REQUEST_DENIED_ERROR("P0920", "Request blocked by security rules. Please consult API documentation for more information.");
+
+        private String value;
+        private String format;
+
+        Code(String value, String format) {
+            this.value = value;
+            this.format = format;
+        }
+
+        public String value() {
+            return value;
+        }
+
+        public String getFormat() {
+            return format;
+        }
+    }
+
+    private String field;
+    private final Code code;
+    private final String description;
+
+    public static AgreementError anAgreementError(Code code, Object... parameters) {
+        return new AgreementError(code, parameters);
+    }
+
+    public static AgreementError anAgreementError(String fieldName, Code code, Object... parameters) {
+        return new AgreementError(fieldName, code, parameters);
+    }
+
+    private AgreementError(Code code, Object... parameters) {
+        this.code = code;
+        this.description = format(code.getFormat(), parameters);
+    }
+
+    private AgreementError(String fieldName, Code code, Object... parameters) {
+        this.field = fieldName;
+        this.code = code;
+        this.description = format(code.getFormat(), concat(fieldName, parameters));
+    }
+
+    @ApiModelProperty(example = "amount")
```

Comment:
Currently we don't have `amount` in the API, should an example be something from agreement API?

Reply:
👍 well spotted, I skim over annotations :D